specutils/io/parsing_utils.py

```diff
@@ -4,10 +4,77 @@
 from astropy.nddata import StdDevUncertainty
 from astropy.utils.exceptions import AstropyUserWarning
 import warnings
+import logging
 
 from specutils.spectra import Spectrum1D
 
-def generic_spectrum_from_table(table, **kwargs):
+
+def spectrum_from_column_mapping(table, column_mapping, wcs=None):
+    """
+    Given a table and a mapping of the table column names to attributes
+    on the Spectrum1D object, parse the information into a Spectrum1D.
+
+    Parameters
+    ----------
+    table : :class:`~astropy.table.Table`
+        The table object returned from parsing the data file.
+    column_mapping : dict
+        A dictionary describing the relation between the file columns
+        and the arguments of the `Spectrum1D` class, along with unit
+        information. The dictionary keys should be the file column names
+        while the values should be a two-tuple where the first element is the
+        associated `Spectrum1D` keyword argument, and the second element is the
+        unit for the file column::
+
+            column_mapping = {'FLUX': ('flux', 'Jy')}
+
+    wcs : :class:`~astropy.wcs.WCS` or :class:`gwcs.WCS`
+        WCS object passed to the Spectrum1D initializer.
+    """
+    spec_kwargs = {}
+
+    # Associate columns of the file with the appropriate spectrum1d arguments
+    for col_name, (kwarg_name, cm_unit) in column_mapping.items():
+        # If the table object couldn't parse any unit information,
+        # fallback to the column mapper defined unit
+        tab_unit = table[col_name].unit
+
+        if tab_unit and cm_unit is not None:
+            # If the table unit is defined, retrieve the quantity array for
+            # the column
+            kwarg_val = u.Quantity(table[col_name], tab_unit)
+
+            # Attempt to convert the table unit to the user-defined unit.
+            logging.debug("Attempting auto-convert of table unit '%s' to "
+                          "user-provided unit '%s'.", tab_unit, cm_unit)
+
+            if cm_unit.physical_type in ['length', 'frequency']:
```

Comment:
(minor performance gain)
```suggestion
            if cm_unit.physical_type in ('length', 'frequency'):
```